src/main/java/com/hubspot/guice/transactional/TransactionalDataSource.java

```diff
@@ -22,12 +25,17 @@ public Connection getConnection() throws SQLException {
     if (TransactionalInterceptor.inTransaction()) {
       TransactionalConnection connection = TransactionalInterceptor.getTransaction();
       if (connection == null) {
-        connection = new TransactionalConnection(delegate.getConnection());
+        Connection delegateConnection = delegate.getConnection();
+        ensureDbNamePopulated(delegateConnection);
+        connection = new TransactionalConnection(delegateConnection, getDbName());
         TransactionalInterceptor.setTransaction(connection);
       }
+      throwIfConnectionInvalid(connection);
 
       return connection;
     } else {
+      Connection connection = delegate.getConnection();
+      ensureDbNamePopulated(connection);
       return delegate.getConnection();
```

Comment:
Should we return `connection` rather than calling `delegate.getConnection()` again here?

Reply:
Oof, yes good catch.